npm/pkg/dataplane/dataplane_windows.go

```diff
@@ -375,36 +375,22 @@ func (dp *DataPlane) getLocalPodEndpoints() ([]*hcn.HostComputeEndpoint, error)
 	return epPointers, nil
 }
 
-func removeCommonEndpoints(endpoints, endpointsAttached *[]hcn.HostComputeEndpoint) []hcn.HostComputeEndpoint {
-	// Choose smaller and larger lists based on length for efficiency
-	smallerEndpointsList, largerEndpointsList := endpoints, endpointsAttached
-	if len(*endpoints) > len(*endpointsAttached) {
-		smallerEndpointsList, largerEndpointsList = endpointsAttached, endpoints
-	}
-
-	// Store IDs of smaller list in a map for quick lookup
-	idMap := make(map[string]struct{}, len(*smallerEndpointsList))
-	for i := 0; i < len(*smallerEndpointsList); i++ {
-		ep := &(*smallerEndpointsList)[i]
+func GetUniqueEndpoints(endpoints, endpointsAttached []hcn.HostComputeEndpoint) []hcn.HostComputeEndpoint {
+	// Store IDs of endpoints list in a map for quick lookup
+	idMap := make(map[string]struct{}, len(endpoints))
+	for i := 0; i < len(endpoints); i++ {
+		ep := &(endpoints)[i]
```

Comment:
what does the `&` do here?

Reply:
removed the & -> had added it originally but realized no elements in endpoints directly need to be modified during iteration so don't need the &. Also removed this line completely in replacement of adding for range loop